pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go

```diff
@@ -180,49 +195,77 @@ func (s *StackConfigPolicyStatus) setReadyCount() {
 	s.ReadyCount = fmt.Sprintf("%d/%d", s.Ready, s.Resources)
 }
 
-func (s *StackConfigPolicyStatus) AddPolicyErrorFor(resource types.NamespacedName, phase PolicyPhase, msg string) error {
-	if _, ok := s.ResourcesStatuses[resource.String()]; ok {
+func (s *StackConfigPolicyStatus) AddPolicyErrorFor(resource types.NamespacedName, phase PolicyPhase, msg string, resourceType ResourceType) error {
+	if _, ok := s.ResourcesStatuses[s.getResourceStatusKey(resource, resourceType)]; ok {
 		return fmt.Errorf("policy error already exists for resource %q", resource)
 	}
-	s.ResourcesStatuses[resource.String()] = ResourcePolicyStatus{
-		Phase: phase,
-		Error: PolicyStatusError{Message: msg},
+	resourcePolicyStatus := ResourcePolicyStatus{
+		ResourceType: resourceType,
 	}
+	switch resourceType {
+	case ElasticsearchResourceType:
+		resourcePolicyStatus.ElasticsearchStatus = ElasticsearchPolicyStatus{
+			Phase: phase,
+			Error: PolicyStatusError{Message: msg},
+		}
+	case KibanaResourceType:
+		resourcePolicyStatus.KibanaStatus = KibanaPolicyStatus{
```

Comment:
It feels like we are missing an interface or common struct that would avoid this code duplication.